forum/models/threads.py

```diff
@@ -92,115 +84,62 @@ def insert(
             "context": context,
             "comment_count": 0,
             "at_position_list": [],
-            "title": title,
-            "body": body,
-            "course_id": course_id,
-            "commentable_id": commentable_id,
+            "title": kwargs.get("title", ""),
```

Comment:
These are the required fields. We can add validations at the top instead of setting the default value to "". The same applies to all other definitions in other classes.

Reply:
These validations can be done in views before accessing the models, this way we'll have all validations at one place as discussed with regis.